src/states/common/base.rs

```diff
@@ -270,29 +265,50 @@ pub trait Base: Display {
         self.full_id().public_id().name()
     }
 
+    fn our_connection_info(&mut self) -> Result<NodeInfo, RoutingError> {
+        self.network_service_mut()
+            .our_connection_info()
+            .map_err(|err| {
+                debug!(
+                    "{} - Failed to retrieve our connection info: {:?}",
+                    self, err
+                );
+                err.into()
+            })
+    }
+
     fn close_group(&self, _name: XorName, _count: usize) -> Option<Vec<XorName>> {
         None
     }
 
     fn send_direct_message(&mut self, dst_id: &PublicId, content: DirectMessage) {
-        let message = match SignedDirectMessage::new(content, self.full_id()) {
-            Ok(message) => message,
-            Err(err) => {
-                error!("{} - Failed to create DirectMessage: {:?}", self, err);
-                return;
-            }
+        let message = if let Ok(message) = self.to_signed_direct_message(content) {
+            message
+        } else {
+            return;
         };
 
-        self.send_message(dst_id, Message::Direct(message));
+        self.send_message(dst_id, message)
     }
 
     fn send_message(&mut self, dst_id: &PublicId, message: Message) {
-        let priority = message.priority();
+        let peer = match self.peer_map().get(dst_id) {
+            Some(peer) => peer.clone(),
+            None => {
+                error!(
+                    "{} - Failed to send message to {:?} - not connected",
+                    self, dst_id
+                );
+                return;
+            }
+        };
+
+        self.send_message_over_network(peer, message);
+    }
 
+    fn send_message_over_network(&mut self, dst: ConnectionInfo, message: Message) {
         match to_network_bytes(message) {
-            Ok(bytes) => {
-                self.send_or_drop(dst_id, bytes, priority);
-            }
+            Ok(bytes) => self.network_service_mut().send(dst, bytes),
```

Comment:
Doesn't `send` return any kind of result? If not, it might cause trouble for RMD - the plan was to depend on whether sending a message was successful quite a bit (and resend to someone else if it wasn't).

Reply:
It doesn't, but if the message fails to be sent, we might receive `UnsentUserMessage` event from quic-p2p later. In this PR we are ignoring such event but there is a `TODO` to address it later (and we have a [github issue](https://github.com/maidsafe/routing/issues/1684) for it too).